backend/sirius-components-diagrams-tests/src/main/java/org/eclipse/sirius/components/diagrams/tests/builder/TestLayoutDiagramBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.tests.builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.diagrams.Edge;
+import org.eclipse.sirius.components.diagrams.Node;
+import org.eclipse.sirius.components.diagrams.Position;
+import org.eclipse.sirius.components.diagrams.Size;
+import org.eclipse.sirius.components.diagrams.tests.builder.edge.EdgeBuilder;
+import org.eclipse.sirius.components.diagrams.tests.builder.node.NodesBuilder;
+
+/**
+ * Used to build a diagram.
+ *
+ * @author gcoutable
+ */
+public final class TestLayoutDiagramBuilder {
+
+    public static final UUID NODE_DESCRIPTION_ID = UUID.nameUUIDFromBytes("NODE_DESCRIPTION_ID".getBytes()); //$NON-NLS-1$
+
+    public static final UUID CHILD_NODE_DESCRIPTION_ID = UUID.nameUUIDFromBytes("CHILD_NODE_DESCRIPTION_ID".getBytes()); //$NON-NLS-1$
+
+    public static final UUID EDGE_DESCRIPTION_ID = UUID.nameUUIDFromBytes("EDGE_DESCRIPTION_ID".getBytes()); //$NON-NLS-1$
+
+    public static final UUID DIAGRAM_DESCRIPTION_ID = UUID.nameUUIDFromBytes("DIAGRAM_DESCRIPTION_ID".getBytes()); //$NON-NLS-1$
+
+    private String diagramId;
+
+    private NodesBuilder<TestLayoutDiagramBuilder> nodesBuilder;
+
+    private List<EdgeBuilder> edgeBuilders = new ArrayList<>();
```

Comment:
Why the difference in the approach used to create nodes and edges?

Reply:
Because in the example you gave me there was no `edges()`. Also, since there is only one kind of edge supported now, I thought that was pointless to define this API with :
```
diagram()
  .nodes()
    ...
  .edges()
    .edges("")
    .edges("")
```
But if you think that's better with `edges()` before all edges, I can add it.

Reply:
No need to change that, it's a good reason indeed. I did not have that difference in mind.